### PR TITLE
165321218 refactoring user

### PR DESCRIPTION
**Pull request by stormrubio3000**

### CODE REVIEW

File: src/ServiceHub.User.Context/Repositories/UserRepository.cs [hunk 1]
```diff
@@ -28,9 +31,10 @@ public UserRepository(IMongoCollection<Models.User> mongoCollection, LastModifie
         /// Insert new user into the data source.
         /// </summary>
         /// <param name="user">The user to insert.</param>
-        public async Task InsertAsync(Models.User user)
+        public async Task InsertAsync(UserModel user)
         {
-            await _users.InsertOneAsync(user);
+            Models.User contextUser = UserModelMapper.LibraryToContext(user);
+            await _users.InsertOneAsync(contextUser);
             lmt.LastModified = DateTime.Now;
```
Comment (c-troha) [inline src/ServiceHub.User.Context/Repositories/UserRepository.cs]: Remove whitespace

File: src/ServiceHub.User.Service/Controllers/UserController.cs [hunk 2]
```diff
@@ -173,23 +179,16 @@ public async Task<IActionResult> GetByType([FromBody] string type)
...
-                var contextUsers = users.Where(x => string.Equals(x.Type, type, StringComparison.OrdinalIgnoreCase)).ToList();
-                List<lib.User> libraryUsers = UserModelMapper.List_ContextToLibrary(contextUsers);
-
-                if (contextUsers.Count > 0 && libraryUsers.Count == 0)
-                {
-                    // respond with null in case of error for compatibility with earlier behavior
-                    libraryUsers = null;
-                }
+                List<UserModel> users = await _userRepository.GetAsync();
+                List<UserModel> libraryUsers = users.Where(x => string.Equals(x.Type, type, StringComparison.OrdinalIgnoreCase)).ToList();
```
Comment (mattvalencia) [inline src/ServiceHub.User.Service/Controllers/UserController.cs]: If we're refactoring, we should probably be moving this type of logic into a different method in the repo.

File: src/ServiceHub.User.Service/Controllers/UserController.cs [hunk 3]
```diff
@@ -308,32 +292,25 @@ public async Task<IActionResult> Put([FromBody] lib.User user)
...
-        public async Task<IActionResult> Post([FromBody] lib.User user)
+        public async Task<IActionResult> Post([FromBody] lib.UserModel user)
         {
             try
             {
-                if (user == null)
+                if (user == null || !user.Validate())
                 {
                     _logger.LogError("User is null.");
                     return BadRequest("Invalid user: User is null");
```
Comment (mattvalencia) [inline src/ServiceHub.User.Service/Controllers/UserController.cs]: If we change the condition, we should change the error message as well.

File: src/ServiceHub.User.Testing/Service/UserControllerPassedQueueTests.cs [hunk 4]
```diff
@@ -728,32 +728,31 @@ public async Task Post_InvalidUser_Return400()
...
 
             var queue = new Mock<IQueueClient>();
             queue.Setup(x => x.SendAsync(It.IsAny<Message>())).Returns(Task.CompletedTask).Verifiable();
 
             var c = new UserController(mockRepo.Object, new LastModifiedTracker(), logger, queue.Object);
 
-            IActionResult result = await c.Post(UserModelMapper.ContextToLibrary(contextUser));
-
+            IActionResult result;            
+            result = await c.Post(libraryUser);
```
Comment (mattvalencia) [inline src/ServiceHub.User.Testing/Service/UserControllerPassedQueueTests.cs]: Why is this two separate lines?

File: src/ServiceHub.User.Testing/Service/UserControllerTests.cs [hunk 5]
```diff
@@ -418,6 +419,8 @@ public async Task PutInvalidAddress_Returns400()
 
             var controller = new UserController(mockRepo.Object, new LastModifiedTracker(), logger);
 
+            if (!user.Validate())
+                user = null;
```
Comment (mattvalencia) [inline src/ServiceHub.User.Testing/Service/UserControllerTests.cs]: This has gotta be wrong. If passing an invalid user isn't returning the correct response, either the method isn't working or the test was designed incorrectly, but we shouldn't change it to basically copy the other test for user being null.

**Pull request merged**